Source/WTF/wtf/FixedBitVector.h

```diff
@@ -0,0 +1,238 @@
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/Assertions.h>
+#include <wtf/Atomics.h>
+#include <wtf/BitVector.h>
+#include <wtf/HashTraits.h>
+#include <wtf/PrintStream.h>
+#include <wtf/StdIntExtras.h>
+#include <wtf/StdLibExtras.h>
+
+namespace WTF {
+
+class FixedBitVector final {
+    WTF_MAKE_FAST_ALLOCATED;
+    using WordType = uintptr_t;
+
+public:
+    FixedBitVector() = default;
+
+    FixedBitVector(size_t size)
+        : m_size(size)
+    {
+        m_bitVector.resize(size);
+    }
+
+    bool concurrentTestAndSet(size_t bitIndex, Dependency = Dependency());
+    bool concurrentTestAndClear(size_t bitIndex, Dependency = Dependency());
+
+    bool testAndSet(size_t bitIndex);
+    bool testAndClear(size_t bitIndex);
+    bool test(size_t bitIndex);
+
+    size_t size() const { return m_size; }
+
+    size_t findBit(size_t startIndex, bool value) const;
+
+    bool operator==(const FixedBitVector&) const;
+
+    unsigned hash() const;
+
+    void dump(PrintStream& out) const;
+
+    class iterator {
+        WTF_MAKE_FAST_ALLOCATED;
+
+    public:
+        iterator()
+            : m_fixedBitVector(nullptr)
+            , m_index(0)
+        {
+        }
+
+        iterator(const FixedBitVector& bitVector, size_t index)
+            : m_fixedBitVector(&bitVector)
+            , m_index(index)
+        {
+        }
+
+        size_t operator*() const { return m_index; }
+
+        iterator& operator++()
+        {
+            m_index = m_fixedBitVector->findBit(m_index + 1, true);
+            return *this;
+        }
+
+        iterator operator++(int)
+        {
+            iterator result = *this;
+            ++(*this);
+            return result;
+        }
+
+        bool isAtEnd() const
+        {
+            return m_index >= m_fixedBitVector->size();
+        }
+
+        bool operator==(const iterator& other) const
+        {
+            return m_index == other.m_index;
+        }
+
+        bool operator!=(const iterator& other) const
+        {
+            return !(*this == other);
+        }
+
+    private:
+        const FixedBitVector* m_fixedBitVector;
+        size_t m_index;
+    };
+
+    // Use this to iterate over set bits.
+    iterator begin() const { return iterator(*this, findBit(0, true)); }
+    iterator end() const { return iterator(*this, size()); }
+
+private:
+    static constexpr unsigned wordSize = sizeof(WordType) * 8;
+    static constexpr WordType one = 1;
+
+    size_t m_size { 0 };
```

Comment:
`BitVector` has 1 neat trick: it encodes all the bits into a `uintptr_t` if the requested number of bits (minus 1) will in the `uintptr_t` itself.  This creates a quirk: if the client requests less bits than that, the `BitVector` still gives it the max inlined bits for the size.  So, one advantage of using a `BitVector` as the backing store is that it makes `FixedBitVector` have the same matching quirk.  This means `FixedBitVector` gets to be a drop in replacement for BitVector, and other than the non-resizable property, we can expect it to work the same way.

But in order to inherit that quirk, we'll have to use `BitVector`'s `size()` tracking instead of rolling our own with this `m_size` field.  Additionally, since `BitVector` already has a means of tracking `size`, it is inefficient to separately track it here.  So, let's remove this `m_size` field.  It is not needed.